client/client.go

```diff
@@ -152,7 +156,7 @@ func (c *Client) Run(ctx context.Context) error {
 	//    It exits when an unexpected message is received from the channel, or when can't be received at all.
 	//    When Run is left, caller stops supervisor, and that allows processSupervisorRequests to exit.
 	// Done() channel is closed when both goroutines exited.
-	oneDone := make(chan struct{}, 2)
+	oneDone := make(chan struct{}, 3)
```

Comment:
Please read the comment above, then update it and code accordingly.

Reply:
I don't understand what do you want. It's your code, not mine. Add code suggestion if you know how it should be.

Reply:
That's _our_ code.